### ISSUE TITLE
SQLITE Db error

### ISSUE THREAD

**Issue by carlosdcuba1** [vladkens/twscrape#2]
Hi, thaks for your job.

I got this error when do this with my credentials

import asyncio
from twscrape import AccountsPool, API, gather
from twscrape.logger import set_log_level

async def main():
    pool = AccountsPool()  # or AccountsPool("path-to.db") - default is `accounts.db` 
    await pool.add_account("", "", "@gmail.com", "")

    # log in to all new accounts
    await pool.login_all()

    api = API(pool)
    async for tweet in api.search("elon musk"):
        print(tweet.id, tweet.user.username, tweet.rawContent)  # tweet is `Tweet` object

if __name__ == "__main__":
    asyncio.run(main())

Traceback (most recent call last):
  File "/root/twscrape/test.py", line 17, in <module>
    asyncio.run(main())
  File "/usr/local/lib/python3.11/asyncio/runners.py", line 190, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/asyncio/base_events.py", line 653, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/root/twscrape/test.py", line 10, in main
    await pool.login_all()
  File "/root/twscrape/twscrape/accounts_pool.py", line 86, in login_all
    await self.login(x)
  File "/root/twscrape/twscrape/accounts_pool.py", line 75, in login
    await self.save(account)
  File "/root/twscrape/twscrape/accounts_pool.py", line 67, in save
    await execute(self._db_file, qs, data)
  File "/root/twscrape/twscrape/db.py", line 16, in wrapper
    raise e
  File "/root/twscrape/twscrape/db.py", line 13, in wrapper
    return await func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/twscrape/twscrape/db.py", line 61, in execute
    await db.execute(qs, params)
  File "/usr/local/lib/python3.11/site-packages/aiosqlite/core.py", line 184, in execute
    cursor = await self._execute(self._conn.execute, sql, parameters)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiosqlite/core.py", line 129, in _execute
    return await future
           ^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiosqlite/core.py", line 102, in run
    result = function()
             ^^^^^^^^^^
sqlite3.OperationalError: near "UPDATE": syntax error

I test this with Debian 11 and Ubuntu 22.04

**Comment by vladkens** [comment]
Hi, @carlosdcuba1. Probably it should be fixed by https://github.com/vladkens/twscrape/pull/1

You can try to install dev version (until I update release in pypi):
```sh
pip install git+https://github.com/vladkens/twscrape.git
```

Or just update sqlite on your laptop (in this case it should work with current version):
```sh
sudo apt install sqlite3

sqlite3 --version
# >=3.35 required
```

**Comment by carlosdcuba1** [comment]
Hi. I install squlite 3.42 and a get this error. I update to the last commit

```
Traceback (most recent call last):
  File "/root/twscrape/test.py", line 17, in <module>
    asyncio.run(main())
  File "/usr/local/lib/python3.11/asyncio/runners.py", line 190, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/asyncio/base_events.py", line 653, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/root/twscrape/test.py", line 13, in main
    async for tweet in api.search("elon musk"):
  File "/root/twscrape/twscrape/api.py", line 61, in search
    async for rep in self.search_raw(q, limit=limit):
  File "/root/twscrape/twscrape/api.py", line 42, in search_raw
    async with QueueClient(self.pool, queue, self.debug) as client:
  File "/root/twscrape/twscrape/queue_client.py", line 36, in __aenter__
    await self._get_ctx()
  File "/root/twscrape/twscrape/queue_client.py", line 55, in _get_ctx
    acc = await self.pool.get_for_queue_or_wait(self.queue)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/twscrape/twscrape/accounts_pool.py", line 127, in get_for_queue_or_wait
    account = await self.get_for_queue(queue)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/twscrape/twscrape/accounts_pool.py", line 122, in get_for_queue
    rs = await fetchone(self._db_file, q2)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/twscrape/twscrape/db.py", line 18, in wrapper
    raise e
  File "/root/twscrape/twscrape/db.py", line 15, in wrapper
    return await func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/twscrape/twscrape/db.py", line 81, in fetchone
    async with db.execute(qs, params) as cur:
  File "/usr/local/lib/python3.11/site-packages/aiosqlite/context.py", line 41, in __aenter__
    self._obj = await self._coro
                ^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiosqlite/core.py", line 184, in execute
    cursor = await self._execute(self._conn.execute, sql, parameters)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiosqlite/core.py", line 129, in _execute
    return await future
           ^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiosqlite/core.py", line 102, in run
    result = function()
             ^^^^^^^^^^
sqlite3.OperationalError: near "RETURNING": syntax error
```

**Comment by vladkens** [comment]
Hi @carlosdcuba1.

Hi. I added the [CLI interface](https://github.com/vladkens/twscrape#cli).

In the CLI you can now see the version of SQLite that is actually used by the python:
```sh
pip install twscrape --upgrade
twscrape version
```

**Comment by variousred** [comment]
i have the fix for the RETURNING issue here:
https://github.com/vladkens/twscrape/pull/1#issuecomment-1558284031

**Comment by variousred** [comment]
> 



> Hi, @carlosdcuba1. Probably it should be fixed by #1
> 
> You can try to install dev version (until I update release in pypi):
> 
> ```shell
> pip install git+https://github.com/vladkens/twscrape.git
> ```
> 
> Or just update sqlite on your laptop (in this case it should work with current version):
> 
> ```shell
> sudo apt install sqlite3
> 
> sqlite3 --version
> # >=3.35 required
> ```

none of the debian package repositories have sqlite3 over 3.34

**Comment by variousred** [comment]
> Hi. I install squlite 3.42 and a get this error. I update to the last commit
> 
> ```
> Traceback (most recent call last):
>   File "/root/twscrape/test.py", line 17, in <module>
>     asyncio.run(main())
>   File "/usr/local/lib/python3.11/asyncio/runners.py", line 190, in run
>     return runner.run(main)
>            ^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/asyncio/runners.py", line 118, in run
>     return self._loop.run_until_complete(task)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/asyncio/base_events.py", line 653, in run_until_complete
>     return future.result()
>            ^^^^^^^^^^^^^^^
>   File "/root/twscrape/test.py", line 13, in main
>     async for tweet in api.search("elon musk"):
>   File "/root/twscrape/twscrape/api.py", line 61, in search
>     async for rep in self.search_raw(q, limit=limit):
>   File "/root/twscrape/twscrape/api.py", line 42, in search_raw
>     async with QueueClient(self.pool, queue, self.debug) as client:
>   File "/root/twscrape/twscrape/queue_client.py", line 36, in __aenter__
>     await self._get_ctx()
>   File "/root/twscrape/twscrape/queue_client.py", line 55, in _get_ctx
>     acc = await self.pool.get_for_queue_or_wait(self.queue)
>           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/root/twscrape/twscrape/accounts_pool.py", line 127, in get_for_queue_or_wait
>     account = await self.get_for_queue(queue)
>               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/root/twscrape/twscrape/accounts_pool.py", line 122, in get_for_queue
>     rs = await fetchone(self._db_file, q2)
>          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/root/twscrape/twscrape/db.py", line 18, in wrapper
>     raise e
>   File "/root/twscrape/twscrape/db.py", line 15, in wrapper
>     return await func(*args, **kwargs)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/root/twscrape/twscrape/db.py", line 81, in fetchone
>     async with db.execute(qs, params) as cur:
>   File "/usr/local/lib/python3.11/site-packages/aiosqlite/context.py", line 41, in __aenter__
>     self._obj = await self._coro
>                 ^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/aiosqlite/core.py", line 184, in execute
>     cursor = await self._execute(self._conn.execute, sql, parameters)
>              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/aiosqlite/core.py", line 129, in _execute
>     return await future
>            ^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/aiosqlite/core.py", line 102, in run
>     result = function()
>              ^^^^^^^^^^
> sqlite3.OperationalError: near "RETURNING": syntax error
> ```

i've found that python has to actually be compiled with the right version of sqlite, so just installing it wont help

**Comment by variousred** [comment]
made a PR for it
https://github.com/vladkens/twscrape/pull/10

**Comment by vladkens** [comment]
Fixed in [`v0.2.2`](https://github.com/vladkens/twscrape/releases/tag/v0.2.2)
